src/hooks/useFetchPriceHistories/useFetchPriceHistories.ts

```diff
@@ -15,6 +19,46 @@ export const useFetchPriceHistories = (assetIds: AssetId[], timeframe: HistoryTi
     dispatch(findPriceHistoryByAssetIds.initiate({ assetIds, timeframe }))
   }, [assetIds, dispatch, timeframe])
 
+  const portfolioLoadingStatus = useAppSelector(selectPortfolioLoadingStatus)
+
+  useQuery({
+    queryKey: ['marketData', assetIds],
+    queryFn: async () => {
+      // Only commented out to make it clear we do NOT want to use RTK options here
+      // We use react-query as a wrapper because it allows us to disable refetch for background tabs
+      // const opts = { subscriptionOptions: { pollingInterval: marketDataPollingInterval } }
+      const timeframe = DEFAULT_HISTORY_TIMEFRAME
+
+      await Promise.all([
+        dispatch(
+          marketApi.endpoints.findPriceHistoryByAssetIds.initiate(
+            {
+              timeframe,
+              assetIds,
+            },
+            // Since we use react-query as a polling wrapper, every initiate call *is* a force refetch here
+            { forceRefetch: true },
+          ),
+        ),
+      ])
+
+      // used to trigger mixpanel init after load of market data
+      dispatch(marketData.actions.setMarketDataLoaded())
+
+      // We *have* to return a value other than undefined from react-query queries, see
+      // https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-4#undefined-is-an-illegal-cache-value-for-successful-queries
+      return null
+    },
+    // once the portfolio is loaded, fetch market data for all portfolio assets
+    // and start refetch timer to keep market data up to date
+    enabled: portfolioLoadingStatus !== 'loading',
+    refetchInterval: marketDataPollingInterval,
+    // Do NOT refetch market data in background to avoid spamming coingecko
+    refetchIntervalInBackground: false,
+    // Do NOT refetch market data on window focus to avoid spamming coingecko
+    refetchOnWindowFocus: false,
+  })
+
```

Comment:
Pretty much a copy and paste job from `<AppContext />` - except this fetches the passed `assetIds`.